frontend/src/pages/modelRegistry/screens/RegisterModel.tsx

```diff
@@ -0,0 +1,319 @@
+import React from 'react';
+import {
+    ActionGroup,
+    Alert,
+    AlertActionCloseButton,
+    Breadcrumb,
+    BreadcrumbItem,
+    Button,
+    Form,
+    FormGroup,
+    FormSection,
+    HelperText,
+    HelperTextItem,
+    InputGroupItem,
+    InputGroupText,
+    Radio,
+    Stack,
+    StackItem,
+    Text,
+    TextInput,
+    Title,
+} from '@patternfly/react-core';
+import { useParams } from 'react-router';
+import { Link } from 'react-router-dom';
+import ApplicationsPage from '~/pages/ApplicationsPage';
+import useRegistrModelData from './useRegistrModelData';
+
+const RegisterModel: React.FC = () => {
+    const [
+        {
+            modelRegistryName,
+            modelName,
+            modelDescription,
+            versionName,
+            versionDescription,
+            sourceModelFormat,
+            modelLocationType,
+            modelLocationEndpoint,
+            modelLocationBucket,
+            modelLocationRegion,
+            modelLocationPath,
+            modelLocationURI,
+        },
+        setData,
+        resetData,
+    ] = useRegistrModelData();
+    const [loading, setIsLoading] = React.useState(false);
+    const [error, setError] = React.useState<Error | undefined>(undefined);
+    const { modelRegistry: mrName } = useParams();
+
+    React.useEffect(() => {
+        setData('modelRegistryName', mrName || "");
+    }, [setData]);
+
+    const isDisabled =
+        !modelRegistryName ||
+        !modelName ||
+        !versionName ||
+        loading ||
+        (modelLocationType === 'URI' && !modelLocationURI) ||
+        (modelLocationType === 'Object storage' &&
```

Comment:
Instead of using string literals for this radio button choice, can we create a `ModelLocationType` enum and use that (and change the type to that from `string` in the object's type)? We'll have similar logic in the submit handler and end up duplicating these strings.